cypress/integration/purchase_create_vendor_spec.js

```diff
@@ -0,0 +1,74 @@
+describe('purchase order Test', function() {
+  before(function() {
+    // login before each test
+    cy.loginByForm();
+  });
+
+  const timestamp = new Date().getTime(); // used in the document names, for ordering
+  const vendorValue = `${timestamp} (Cypress Test)`;
+
+  it('Create a vendor with two contacts', function() {
+    cy.visit('/window/123/NEW');
+    // cy.writeIntoStringField('Value', `{selectall}{backspace}${vendorValue}`);
+    cy.writeIntoStringField('CompanyName', vendorValue);
+    cy.writeIntoStringField('Name2', 'CompanyName');
+
+    cy.selectTab('Vendor');
+    cy.selectSingleTabRow();
+
+    cy.openAdvancedEdit();
+    cy.clickOnCheckBox('IsVendor');
+    cy.pressDoneButton();
+
+    // note that the partner needs an address to be eligible in the purchase order without causing an error
+    // TODO: Use proper tab name when rolled out - Kuba
+    // cy.selectTab('C_BPartner_Location');
+    cy.selectTab('Window-123-222');
+    cy.pressAddNewButton();
+    cy.writeIntoStringField('Name', 'Address');
+
+    cy.server()
+    cy.route('POST', '/rest/api/address').as('postAddress')
+    
+    // TODO: extract into a command
+    // Kuba: I'm not sure if this wouldn't be too specific for this use case.
+    cy.get('.form-field-C_Location_ID').find('button').click();
```

Comment:
> I'm not sure if this wouldn't be too specific for this use case.

I think I don't understand.
We have these location widgets in many places, only the filedname (here: `C_Location_ID`) might vary.
And of course the
```JS
 cy.writeIntoStringField('City', 'Cologne')
        .writeIntoCompositeLookupField('C_Country_ID', 'Deu', 'Deutschland');
```
will vary.
I guess in JS we can do closures, right?
Maybe for a test-dev-user it might look like

```JS
cy.editAddressField(
	'C_Location_ID',
	function () {
	    cy.writeIntoStringField('City', 'Cologne')
        cy.writeIntoCompositeLookupField('C_Country_ID', 'Deu', 'Deutschland');
  	}
)
```
(i hope this makes sense and I didn't screw up the syntax!)
And the command would perform all the cy.server, route, emit etc.
wdyt?

Note: we have a similar thing for `M_AttributeSetInstance_ID`, e.g. in order lines. maybe we can even have one command for both cases? but i would also be fine with two commands (and if it's extra efffort, let's do `M_AttributeSetInstance_ID` later..but it will come soon)

Reply:
We're talking two different things here. What I meant is having a custom command for clicking a button. It saves us maybe 20 characters tops when used multiple times. And sometimes we might want to check this button's text for instance. So having this command makes no sense to me and as far as I understood, it's against the recommended best practices.

The rest is fine.